tests/dtm/test_rgw_s3_process_restart.py

```diff
@@ -291,3 +288,211 @@ def test_delete_during_rgw_svc_restart(self):
+                      "Objects")
+        for size in self.test_cfg["size_list"]:
+            resp = self.s3_test_obj.copy_object(source_bucket=bucket_list[0],
+                                                source_object=f"{object_name}_{size}",
+                                                dest_bucket=bucket_list[1],
+                                                dest_object=f"{object_name}_{size}")
+            assert_utils.assert_true(resp[0], resp[1])
+            file_name_copy = "{}{}".format("dtm-test-42253-copy", size)
+            file_path_copy = os.path.join(self.test_dir_path, file_name_copy)
+            resp = self.s3_test_obj.object_download(bucket_name=bucket_list[1],
+                                                    obj_name=f"{object_name}_{size}",
+                                                    file_path=file_path_copy)
+            assert_utils.assert_true(resp[0], resp[1])
+            file_name = "{}{}".format("dtm-test-42253-", size)
+            file_path = os.path.join(self.test_dir_path, file_name)
+            resp = system_utils.validate_checksum(file_path_1=file_path, file_path_2=file_path_copy)
+            assert_utils.assert_true(resp, "Checksum validation Failed.")
+        self.test_completed = True
+        self.log.info("ENDED: Verify copy object after rgw restart using pkill")
+
+    @pytest.mark.lc
+    @pytest.mark.dtm
+    @pytest.mark.tags("TEST-42254")
+    def test_copy_object_during_rgw_restart(self):
+        """Verify copy object during rgw restart using pkill."""
+        self.log.info("STARTED: Verify copy object during rgw restart using pkill")
+        object_name = 'object-test-42254'
+        workload = dict()
+        obj_list = list()
+        bucket_list = list()
+        que = multiprocessing.Queue()
+
+        self.log.info("Step 1: Start write Operations :")
+        for i in range(0, 2):
+            bucket_name = f"bucket-test-42254-{i}"
+            resp = self.s3_test_obj.create_bucket(bucket_name=bucket_name)
+            assert_utils.assert_true(resp[0], resp[1])
+            bucket_list.append(bucket_name)
+        for size in self.test_cfg["size_list"]:
+            file_name = "{}{}".format("dtm-test-42254-", size)
+            file_path = os.path.join(self.test_dir_path, file_name)
+            system_utils.create_file(file_path, size)
+            resp = self.s3_test_obj.put_object(bucket_list[0], f"{object_name}_{size}",
+                                               file_path)
+            obj_list.append(f"{object_name}_{size}")
+            assert_utils.assert_true(resp[0], resp[1])
+
+        self.log.info("Step 2: Perform Copy object to bucket-2 in background")
+        workload["source_bucket"] = bucket_list[0]
+        workload["dest_bucket"] = bucket_list[1]
+        workload["obj_list"] = obj_list
+        proc_cp_op = multiprocessing.Process(target=self.dtm_obj.perform_copy_objects,
```

Comment:
may have asked this before but existing ha lib copy object is not useful?

Reply:
I guess we could use them here also, but need to check with @kulrk for more details, as both copy object tests are written by him. I just moved these two tests from other file to this.